erpnext/accounts/page/bank_reconciliation/bank_reconciliation.py

```diff
@@ -18,6 +18,10 @@ def reconcile(bank_transaction, payment_doctype, payment_name):
 	account = frappe.db.get_value("Bank Account", transaction.bank_account, "account")
 	gl_entry = frappe.get_doc("GL Entry", dict(account=account, voucher_type=payment_doctype, voucher_no=payment_name))
 
+	if payment_entry.unallocated_amount > transaction.unallocated_amount:
```

Comment:
only check this condition if `payment_doctype ` is "Payment Entry" as other doctypes won't have `unallocated_amount`